src/Microsoft.Dnx.Runtime/Project.cs

```diff
@@ -23,19 +19,19 @@ public class Project
 
         public static TypeInformation DefaultCompiler = DefaultRuntimeCompiler;
 
-        private readonly Dictionary<FrameworkName, TargetFrameworkInformation> _targetFrameworks = new Dictionary<FrameworkName, TargetFrameworkInformation>();
-        private readonly Dictionary<FrameworkName, CompilerOptions> _compilationOptions = new Dictionary<FrameworkName, CompilerOptions>();
-        private readonly Dictionary<string, CompilerOptions> _configurations = new Dictionary<string, CompilerOptions>(StringComparer.OrdinalIgnoreCase);
+        // REVIEW: It's kinda hacky making these internal but the reader needs to set them
```

Comment:
Could you put them in a constructor? If you did that I think it'd be OK for the constructor to be public and the grossness would go away.